### PR TITLE
API OTP Context

### PR DESCRIPTION
**Pull request by techknowlogick**
as title

### CODE REVIEW

**Review by jolheiser — COMMENTED** [review]
Looks good so far, just needs the discussed changes from Discord. :+1:

File: modules/context/auth.go [hunk 1]
```diff
@@ -1,10 +1,12 @@
 // Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2019 The Gogs Authors. All rights reserved.
```
Comment (jolheiser) [inline modules/context/auth.go]: Gitea, not Gogs.

**Review by jolheiser — CHANGES_REQUESTED** [review]
One small nit

File: modules/context/auth.go (hunk 1 above)
Comment (techknowlogick) [reply modules/context/auth.go]: 😂

**Review by jolheiser — APPROVED** [review]

File: routers/api/v1/api.go [hunk 2]
```diff
@@ -172,6 +172,9 @@ func reqToken() macaron.Handler {
 		if true == ctx.Data["IsApiToken"] {
 			return
 		}
+		if ctx.Context.IsBasicAuth {
+			ctx.CheckForOTP()
```
Comment (jolheiser) [inline routers/api/v1/api.go:176]: Does this need a return now? Otherwise it will fall through to the next and may write again.

**Review by jolheiser — CHANGES_REQUESTED** [review]

File: routers/api/v1/api.go (hunk 2 above)
Comment (techknowlogick) [reply routers/api/v1/api.go:176]: good catch. I've just added it back in.

**Review by jolheiser — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**